core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -611,6 +611,8 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
         AppInfoParser.unregisterAppInfo(jmxPrefix, config.brokerId.toString)
         shutdownLatch.countDown()
         info("shut down completed")
+      } else {
+        isShuttingDown.set(false)
```

Comment:
@ijuma yea def this could be handled cleaner, I just didn't want to go for a big change since this was (at least for me already a little non trivial to find). If you want I can move to a cleaner synchronized approach for this later? :)